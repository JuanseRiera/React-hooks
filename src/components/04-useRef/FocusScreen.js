import React, { useRef } from "react";
import "./FocusScreen.css";

export const FocusScreen = () => {
  const inputRef = useRef();
  const handleClick = () => {
    inputRef.current.select();
  };

  return (
    <>
      <h1>Focus Screen</h1>
      <hr />

      <input
        ref={inputRef}
        type="text"
        className="form-control mt-3"
        placeholder="Hola"
      />

      <button className="btn btn-secondary mt-3" onClick={handleClick}>
        Focus
      </button>
    </>
  );
};
