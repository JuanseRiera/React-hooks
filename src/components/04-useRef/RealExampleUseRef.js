import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";
import "./FocusScreen.css";

export const RealExampleUseRef = () => {
  const [mostrar, setmostrar] = useState(false);

  return (
    <>
      <h1>Real example use ref</h1>
      <hr />

      {mostrar && (
        <div className="p-5">
          <MultipleCustomHooks />
        </div>
      )}

      <div className="mt-5">
        <button
          className="btn btn-info"
          onClick={() => {
            setmostrar(!mostrar);
          }}
        >
          Mostrar
        </button>
      </div>
    </>
  );
};
