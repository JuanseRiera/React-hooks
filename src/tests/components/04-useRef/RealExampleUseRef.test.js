import { shallow } from "enzyme";
import { RealExampleUseRef } from "../../../components/04-useRef/RealExampleUseRef";
describe("Prubas del componente <RealExampleUseRef/>", () => {
  test("deberia cargar el componente correctamente", () => {
    const wrapper = shallow(<RealExampleUseRef />);
    expect(wrapper).toMatchSnapshot();
  });

  test("deberia cargar el componente <MultipleCustomHooks/> correctamente", () => {
    const wrapper = shallow(<RealExampleUseRef />);
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
