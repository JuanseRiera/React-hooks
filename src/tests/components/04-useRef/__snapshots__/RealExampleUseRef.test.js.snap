// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubas del componente <RealExampleUseRef/> deberia cargar el componente correctamente 1`] = `
<Fragment>
  <h1>
    Real example use ref
  </h1>
  <hr />
  <div
    className="mt-5"
  >
    <button
      className="btn btn-info"
      onClick={[Function]}
    >
      Mostrar
    </button>
  </div>
</Fragment>
`;
